service/src/IOGroup.cpp

```diff
@@ -99,7 +99,8 @@ namespace geopm
 #ifdef GEOPM_ENABLE_RAWMSR
             // Only use /dev/cpu/*/msr if configured with
             // --enable-rawmsr and msr-safe driver is not available.
-            bool use_msr_safe = (access("/dev/cpu/msr_batch", R_OK | W_OK) == 0);
+            bool use_msr_safe = (access("/dev/cpu/msr_batch", R_OK | W_OK) == 0 &&
+                                 std::getenv("GEOPM_DISABLE_MSR_SAFE") == nullptr);
```

Comment:
From Helper.[ch]pp:
```suggestion
                                 geopm::get_env("GEOPM_DISABLE_MSR_SAFE").empty);
```
Aside from the LevelZero.cpp checking ```ZE_AFFINITY_MASK``` directly with ```getenv``` (which imo is another issue we should correct), there are  *no* direct usages of ```getenv``` anywhere in the service code.

Reply:
We really want to know if the environment variable is set or not, we don't care about its contents.  In this way, the geopm::get_env() is ambiguous.  Since we are not using the C string in this case, I think the direct use of std::getenv() is valid.  If a user simply called `export GEOPM_DISABLE_MSR_SAFE` the original code would disable msr-safe use, but the proposed code would not.

Reply:
SGTM, then this PR needs to include some documentation about the expected usage of ```GEOPM_DISABLE_MSR_SAFE```.  IIRC we don't have other env vars that are exported without a value set, but I may be mistaken.